prdoc/pr_6446.prdoc

```diff
@@ -0,0 +1,14 @@
+title: Make pallet-recovery supports `BlockNumberProvider`
+doc:
+- audience: Runtime Dev
+  description: |-
+      This PR enhances the recovery pallet by enabling it to define the source of the block number.
+
+      A migration is required for `Recoverable` and `ActiveRecoveries` storage items in cases where this pallet has been deployed on a parachain, and its block number type differs from that of the relaychain.
```

Comment:
This is not true, if it is on a parachain, it needs a migration if it differs from system.

This doc should say that pallet recovery now allows to configure the block provider to be used in this pallet. This block is used for the delay of the recovery.
So it has a new associated type on the `Config` trait: `BlockNumberProvider`, this can be set to `System` to keep the behavior same as before, or other block number provider such as `RelayChain`.
In case the block provider is set to a different value than `System` then a migration is required for `Recoverable` and `ActiveRecoveries` storage items.

Reply:
Thank you for your explaination. I'm still puzzled as to why it utilizes a delay in the pallet. My understanding is that using the block number from the relay chain aims to enhance accuracy.

Reply:
what do you mean by "it utilizes a delay in the pallet" ?

The delay is used so that the recoverable account can close an unwanted active recovery.

In general I think chains have to set what is good for them. Like if I create a chain that creates one block every 100000 relay chain block, I might not want to use the relay chain block number.